fofpetaio.c

```diff
@@ -77,13 +73,27 @@ void fof_save_particles(int num) {
     MPI_Barrier(MPI_COMM_WORLD);
     fof_write_header(&bf); 
 
+    int stype = -1;
+    int * argind = NULL;
+    int NumPIG = 0;
+
     for(i = 0; i < IOTable.used; i ++) {
         /* only process the particle blocks */
         char blockname[128];
         int ptype = IOTable.ent[i].ptype;
         BigArray array = {0};
         if(ptype < 6 && ptype >= 0) {
+            walltime_measure("/FOF/IO/Write");
             sprintf(blockname, "%d/%s", ptype, IOTable.ent[i].name);
+            if(ptype != stype) {
```

Comment:
Note that the entries in IOTable are sorted by particle type, so we would only see each ptype once - the LRU cache would be rebuilt strictly once.

But I had forgotten that the particles are sorted by type at this point for efficiency anyway, so indeed we can use a shifted loop. I will add a comment too that we are relying on this.